packages/cdk/lib/cloudwatch-logs-management.ts

```diff
@@ -1,11 +1,214 @@
+
+		super(scope, id, {
+			...props,
+			stack,
+			stage: 'PROD',
+			env: {
+				region: 'eu-west-1',
+			},
+		});
+
+		const { region, account } = this;
+
+		const kinesisStreamArn: string = new GuStringParameter(
+			this,
+			'KinesisStreamArn',
+			{
+				fromSSM: true,
+				default: '/account/services/logging.stream',
+				description: 'The ARN (not name) of the kinesis stream to ship logs to',
+			},
+		).valueAsString;
+
+		const structuredFieldsBucket = new GuS3Bucket(
+			this,
+			'StructuredFieldsBucket',
+			{ app: 'cloudwatch-logs-management' },
+		);
+
+		this.overrideLogicalId(structuredFieldsBucket, {
+			logicalId: 'StructuredFieldsBucket',
+			reason: 'Migrating from YAML',
+		});
+
+		const setRetentionLambda = new GuScheduledLambda(this, 'set-retention', {
+			app: 'set-retention',
+			runtime: Runtime.NODEJS_16_X,
+			fileName: 'cloudwatch-logs-management.zip',
+			handler: 'app.setRetention',
+			rules: [{ schedule: Schedule.rate(Duration.hours(1)) }],
+			monitoringConfiguration: { noMonitoring: true },
+			environment: {
+				RETENTION_IN_DAYS: retentionInDays.toString(),
+			},
+			timeout: Duration.minutes(1),
+		});
+
+		this.overrideLogicalId(setRetentionLambda, {
+			logicalId: 'SetRetentionFunc',
+			reason: 'Migrating from YAML',
+		});
+
+		const setRetentionPolicy = new ManagedPolicy(this, 'SetRetentionPolicy', {
+			statements: [
+				new PolicyStatement({
+					effect: Effect.ALLOW,
+					actions: ['logs:DescribeLogGroups', 'logs:PutRetentionPolicy'],
+					resources: [`arn:aws:logs:${region}:${account}:log-group:*`],
+				}),
+			],
+		});
+		setRetentionLambda.role?.addManagedPolicy(setRetentionPolicy);
+
+		const shipLogEntriesLambda = new GuLambdaFunction(
+			this,
+			'ship-log-entries',
+			{
+				app: 'ship-log-entries',
+				runtime: Runtime.NODEJS_16_X,
+				fileName: 'cloudwatch-logs-management.zip',
+				handler: 'shipLogEntries.shipLogEntries',
+				timeout: Duration.seconds(5),
+				environment: {
+					LOG_KINESIS_STREAM: kinesisStreamArn,
+					STRUCTURED_DATA_BUCKET: structuredFieldsBucket.bucketName,
+				},
+			},
+		);
+
+		this.overrideLogicalId(shipLogEntriesLambda, {
+			logicalId: 'ShipLogEntriesFunc',
+			reason: 'Migrating from YAML',
+		});
+
+		shipLogEntriesLambda.addPermission('ShipLogEntriesPermission', {
+			principal: new ServicePrincipal(`logs.${region}.amazonaws.com`),
+			sourceAccount: this.account,
+		});
+
+		const shipLogEntriesPolicies = [
+			new ManagedPolicy(this, 'ShipLogEntriesPolicy', {
+				statements: [
+					new PolicyStatement({
+						effect: Effect.ALLOW,
+						actions: ['kinesis:PutRecords'],
+						resources: [kinesisStreamArn],
+					}),
+					new PolicyStatement({
+						effect: Effect.ALLOW,
+						actions: ['s3:GetObject'],
+						resources: [`${structuredFieldsBucket.bucketArn}/*`],
+					}),
+				],
+			}),
+			new ManagedPolicy(this, 'DisableCloudWatchLoggingPolicy', {
```

Comment:
It's probably worth preserving [this comment](https://github.com/guardian/cloudwatch-logs-management/blob/1f7271512ba39dd33fc94f67aad2940f5c64478a/template.yaml#L205-L210)?

Reply:
Done!